### PR TITLE
serveit: use ruby 3.3 and add webrick dep

### PR DESCRIPTION
**Pull request by chenrui333**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

https://github.com/Homebrew/homebrew-core/actions/runs/10223865606/job/28290866353

### CODE REVIEW

**Comment by chenrui333** [comment]
```
==> Testing serveit
  webrick is not part of the default gems since Ruby 3.0.0. Install webrick from RubyGems.
  <internal:/home/linuxbrew/.linuxbrew/Cellar/ruby/3.3.4/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require': cannot load such file -- webrick (LoadError)
  	from <internal:/home/linuxbrew/.linuxbrew/Cellar/ruby/3.3.4/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
  	from /home/linuxbrew/.linuxbrew/Cellar/serveit/0.0.3_1/bin/serveit:4:in `<main>'
  ==> curl localhost:4[51](https://github.com/Homebrew/homebrew-core/actions/runs/10227809497/job/28299723089?pr=179624#step:4:52)89
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
  
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  curl: (7) Failed to connect to localhost port 45189 after 0 ms: Connection refused
```

**Comment by chenrui333** [comment]
moved to use newer ruby

File: Formula/s/serveit.rb [hunk 1]
```diff
@@ -11,8 +11,25 @@ class Serveit < Formula
     sha256 cellar: :any_skip_relocation, all: "e17a21fd5705de1b2e96656ad703f51c8f1781601f295ff7028cfc859bd0ca44"
   end
 
+  depends_on "ruby"
```
Comment (carlocab) [inline Formula/s/serveit.rb:14]: Does `uses_from_macos "ruby"` not work?

Comment (chenrui333) [reply Formula/s/serveit.rb:14]: it wont work for linux build as it would use ruby 3.3, so I just switched all builds to use ruby 3.3 as result

Comment (branchvincent) [reply Formula/s/serveit.rb:14]: can we keep `uses_from_macos` and just scope the gem resource to `on_linux`?

Comment (chenrui333) [reply Formula/s/serveit.rb:14]: yeah, I can try that.

**Review by p-linnane — APPROVED** [review]

**Pull request merged**